File: src/0874-backspace-string-compare/solution.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var backspaceCompare = function(s, t) {
  const prep = function(a) {
    let b = [];
    for(let i = 0; i < a.length; i++) {
      if (a.charAt(i) !== '#')
        b.push(a.charAt(i))
      else
        b.pop();
    }
    return b.join('');
  }

  s = prep(s);
  t = prep(t);

  return s === t;
};

File: src/0874-backspace-string-compare/solution.py
class Solution:
    def typed(self, s: str) -> str:
        a = list()
        for c in s:
            if c != "#": a.append(c)
            elif len(a) > 0: a.pop()
        return "".join(a)

    def backspaceCompare(self, s: str, t: str) -> bool:
        return self.typed(s) == self.typed(t)
        
